lib/lockbox/model.rb

```diff
@@ -143,7 +143,31 @@ def attributes
                   # it is possible that the encrypted attribute is not loaded, eg.
                   # if the record was fetched partially (`User.select(:id).first`).
                   # accessing a not loaded attribute raises an `ActiveModel::MissingAttributeError`.
-                  send(lockbox_attribute[:attribute]) if has_attribute?(lockbox_attribute[:encrypted_attribute])
+                  encrypted_attribute = lockbox_attribute[:encrypted_attribute]
+                  if has_attribute?(encrypted_attribute)
+                    send(lockbox_attribute[:attribute])
+                    next
+                  end
+
+                  # encrypted attribute could be inside an ActiveRecord::Store
+                  # which doesn't expose accessors of store as attributes, so
+                  # check if attribute for store is loaded
+                  store, _ = self.class.stored_attributes.detect do |_, accessors|
+                    accessors.map(&:to_sym).include?(encrypted_attribute.to_sym)
+                  end
+                  if store
+                    # if entire store is encrypted, check if encrypted attribute
+                    # for store is loaded
+                    store_encrypted_attribute = self.class.lockbox_attributes.dig(
+                      store.to_sym, :encrypted_attribute
+                    )
+                    if (
+                      (store_encrypted_attribute && has_attribute?(store_encrypted_attribute)) ||
+                      (store_encrypted_attribute.nil? && has_attribute?(store))
```

Comment:
One new issue introduced by this fix is the `#inspect` method won't replace encrypted ciphertext attributes if they are inside of a store that isn't encrypted. See `password_ciphertext` in example below using `User` from tests
```
[1] pry> user = User.create!(email: "test@example.org", password: 'asd');
[2] pry> user.inspect
=> "#<User id: 23, name: nil, document: nil, documents: nil, email: [FILTERED], phone: nil, properties: nil, properties2: nil, settings: {}, settings2: nil, messages: [], messages2: nil, country: nil, country2: nil, active: nil, active2: nil, born_on: nil, born_on2: nil, signed_at: nil, signed_at2: nil, opens_at: nil, opens_at2: nil, sign_in_count: nil, sign_in_count2: nil, latitude: nil, latitude2: nil, longitude: nil, longitude2: nil, video: nil, video2: nil, data: nil, data2: nil, info: {}, info2: nil, credentials: {\"password_ciphertext\"=>\"lR9U5yLXC0HWGoQlqvi8zP+ruoOihg0bwBjD4xmSDA==\"}, credentials2: nil, configuration: \"\", configuration2: nil, coordinates: [], coordinates2: nil, config: \"\", config2: nil, migrated_conf: nil, city: nil, ssn: nil, region: nil, state: nil, photo_data: nil, conf: \"\">"
```

Reply:
Confirmed that one can nest `ActiveRecord::Store` as many times as they want.
```
class User < ActiveRecord::Base
  store :credentials, accessors: [:store1], coder: JSON
  store :store1, accessors: [:store2], coder: JSON
  store :store2, accessors: [:store3], coder: JSON
  store :store3, accessors: [:foo_ciphertext], coder: JSON
  has_encrypted :foo
end
```
and here is an example of what `#attributes` and `#inspect` looks like
```
[1] pry(#<ModelTest>)> user = User.create!(foo: "hello world!");
[2] pry(#<ModelTest>)> puts JSON.pretty_generate(user.reload.attributes.slice('credentials', 'store1', 'store2', 'store3', 'foo'))
{
  "credentials": {
    "store1": {
      "store2": {
        "store3": {
          "foo_ciphertext": "28IOE0QYlTprQQf6B58c0nwKQF3Co0z/Mms3F2FpDe5PCdgxB0zDQw=="
        }
      }
    }
  },
  "store1": null,
  "store2": null,
  "store3": null,
  "foo": "hello world!"
}
[3] pry(#<ModelTest>)> user.inspect
=> "#<User id: 62, name: nil, document: nil, documents: nil, email: nil, phone: nil, properties: nil, properties2: nil, settings: {}, settings2: nil, messages: [], messages2: nil, country: nil, country2: nil, active: nil, active2: nil, born_on: nil, born_on2: nil, signed_at: nil, signed_at2: nil, opens_at: nil, opens_at2: nil, sign_in_count: nil, sign_in_count2: nil, latitude: nil, latitude2: nil, longitude: nil, longitude2_ciphertext: nil, video: nil, video2: nil, data: nil, data2: nil, info: {}, info2: nil, credentials: {\"store1\"=>{\"store2\"=>{\"store3\"=>{\"foo_ciphertext\"=>\"28IOE0QYlTprQQf6B58c0nwKQF3Co0z/Mms3F2FpDe5PCdgxB0zDQw==\"}}}}, credentials2: nil, configuration: nil, configuration2: nil, coordinates: [], coordinates2: nil, config: nil, config2: nil, migrated_conf: nil, city: nil, ssn: nil, region_ciphertext: nil, state: nil, photo_data: nil, store1: nil, store2: nil, store3: nil, conf: nil>"
```
A little unfortunate that `store1`, `store2,` and `store3`  are all `nil` in those outputs when they do have the expected values when calling their accessors.